### PR TITLE
Routing: infrastructure and ol logic

### PR DESCRIPTION
**Pull request by taulinger**
- [x]  add basic infrastructure
- [x] port ol logic from bav3

Excluded in this branch:
- <s>highlighting of segments</s> (included!)
- deleting  waypoints
- adding waypoints by clicking on the map

### CODE REVIEW

File: src/services/provider/routingCategories.provider.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/**
+ * @module services/provider/routingCategories_provider
+ */
+import { $injector } from '../../injection/index';
+
+/**
+ * Bvv specific implementation of {@link module:services/RoutingService~routingCategoriesProvider}
+ * @function
+ * @type {module:services/FeedbackService~feedbackStorageProvider}
```
Comment (thiloSchlemmer) [inline src/services/provider/routingCategories.provider.js:9]: doc typo
should be `* @type {module:services/RoutingService~routingCategoriesProvider}`

File: src/modules/olMap/handler/routing/OlRoutingHandler.js [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+				map.getTarget().style.cursor = 'grab';
+			} else {
+				map.getTarget().style.cursor = 'pointer';
+			}
+		};
+		return (event) => {
+			this._helpTooltip.deactivate();
+
+			if (!event?.dragging) {
+				// $(element).popover('destroy');
```
Comment (thiloSchlemmer) [inline src/modules/olMap/handler/routing/OlRoutingHandler.js:160]: if this is a relevant hint from the porting v3 code, then a meaningful rewrite would be nice, to get a quicker understanding

File: src/modules/olMap/handler/routing/OlRoutingHandler.js [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+
+				const pixel = map.getEventPixel(event.originalEvent);
+				const hit = map.getFeaturesAtPixel(
+					pixel,
+					this._getPointerMoveGetFeaturesAtPixelOptions(interactionLayer, alternativeRouteLayer, routeLayerCopy)
```
Comment (thiloSchlemmer) [inline src/modules/olMap/handler/routing/OlRoutingHandler.js:165]: a hard readable method name...but a better suggestion is even harder :D  
How about 
```suggestion
					this._getFeaturesAtPixelOptionsForPointerMove(interactionLayer, alternativeRouteLayer, routeLayerCopy)
```

File: src/modules/olMap/handler/routing/OlRoutingHandler.js [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+	 *
+	 * @param {object} routeGeometry
+	 * @returns features
+	 */
+	_splitRouteByIntermediatePoints(routeGeometry) {
+		const determineClosestCoordinate = (coords, closest) => {
+			let dist = Number.MAX_VALUE;
+			let index = -1;
+			for (let i = 0; i < coords.length; i++) {
+				// Note: we're calculating the planar distance although we are having coordinates, but it should be precisely enough in this case
```
Comment (thiloSchlemmer) [inline src/modules/olMap/handler/routing/OlRoutingHandler.js:350]: is there 'spherical' missing? Like 'we are having spherical coordinates

File: src/modules/olMap/handler/routing/OlRoutingHandler.js (hunk 2 above)
Comment (taulinger) [reply src/modules/olMap/handler/routing/OlRoutingHandler.js:160]: As discussed, I suggest leaving the hint untouched until the context menu will be called from this place

**Review by thiloSchlemmer — APPROVED** [review]
Nice!

**Pull request merged**